gridRow[] grid
pointData[] path
int16 pathLength